### ISSUE TITLE
Standardise Docstrings to Reflect Promise Returns as {Promise<Object>} Format

### ISSUE THREAD

**Issue by Beckyrose200** [DEFRA/water-abstraction-team#110]
Lately, while working, we've encountered Sonar Cloud flagging 'redundant awaits' in our codebase. Despite this, these awaits are crucial as our functions operate asynchronously. This was confusing for us because it seemed like a false positive in Sonar Cloud. 

Upon investigation, it became clear that Sonar Cloud detects issues with promises and awaiting functions by parsing function doctrine comments. Previously, our coding conventions outlined that docstring comments should only specify return values, omitting any details about promises or asynchronous functions. 

We've now realised the necessity of modifying our docstring style comments to incorporate the Promise tag. This adjustment aims to prevent Sonar Cloud from raising such complaints. Temporarily, we've been labelling these occurrences as false positives in Sonar Cloud. However, since updating our coding conventions to include this new promise format we now need to go back into our code base and make all the changes to these comments.

TLDR: To address Sonar Cloud's concerns, we're updating the docstring style comments for asynchronous functions to include the Promise tag.

Example good:

```
/**
 * Fetches SROC return logs.
 *
 * @param {String} licenceRef - The reference of the license that the return log relates to.
 * @param {Object} billingPeriod - An object containing properties `startDate` and `endDate`
 *     representing the period being billed.
 * 
 * @returns {Promise<Object>}
 * /
```

Example bad:

```
/**
 * Fetches SROC return logs.
 *
 * @param {String} licenceRef - The reference of the license that the return log relates to.
 * @param {Object} billingPeriod - An object containing properties `startDate` and `endDate`
 *     representing the period being billed.
 * 
 * @returns {Object}
 * /
```

NOTE: This is only for the system repo where we are using our teams coding conventions.
